### ISSUE TITLE
filetypeinfo failing on subsequent requests

### ISSUE THREAD

**Issue by elschnert** [LarsKoelpin/magic-bytes#27]
I'm using node and express to run some microservices. One of the services needs to get the mime-type from a file and I noticed that the first request performs as expected. Every subsequent (identical) requests didn't get the right mimetype. The input to the function is the same (verified), but the result is just an empty array.
If I use filetypeinfo several times in the same request it works as expected.

I narrowed it down: It seems as the the orginal tree gets modified by accident, and whatever was matched there is removed from the tree, so It can't be matched again.

I have a fix that works for me (PR incoming). I was not able to reproduce this behaviour with jest.

**Comment by LarsKoelpin** [comment]
I merged your PR and published version 1.0.12

**Comment by elschnert** [comment]
You are right, I missed that...

Not working (<1.0.12):
```
const guessedFileTypes = filetype(chunk);
if (guessedFileTypes.length) {
    mimetype = guessedFileTypes.shift()!.mime ?? undefined;
}
```

Working:
```
const guessedFileTypes = filetype(chunk);`
if (guessedFileTypes.length) {`
    mimetype = guessedFileTypes[0].mime ?? undefined;
}
```


Thanks!
